src/_C014.ino

```diff
@@ -67,872 +68,1069 @@ bool CPlugin_014_sendMQTTmsg(String& topic, const char* payload, int& errorCount
-                            case 3: valueName = F("string"); break;
-                            case 4: valueName = F("enum");
-                                    unitName = ExtraTaskSettings.TaskDeviceFormula[varNr];
-                                    break;
-                            case 5: valueName = F("color");
-                                    unitName = F("rgb");
-                                    break;
-                            case 6: valueName = F("color");
-                                    unitName = F("hsv");
-                                    break;
-                          }
-                          CPlugin_014_sendMQTTnode(nodename, deviceName.c_str(), ExtraTaskSettings.TaskDeviceValueNames[varNr], "/$datatype", valueName.c_str(), errorCounter);
-                          if (!unitName.isEmpty()) CPlugin_014_sendMQTTnode(nodename, deviceName.c_str(), ExtraTaskSettings.TaskDeviceValueNames[varNr], "/$format", unitName.c_str(), errorCounter);
-                          nodeCount++;
+                  for (uint8_t varNr = 0; varNr < valueCount; varNr++) {
+                    if (validPluginID_fullcheck(Settings.TaskDeviceNumber[x])) {
+                      if (ExtraTaskSettings.TaskDeviceValueNames[varNr][0] != 0) { // do not send if Value Name is empty!
+                        CPLUGIN_014_addToList(valuesList, ExtraTaskSettings.TaskDeviceValueNames[varNr]);
+
+                        // $settable	Device → Controller	Specifies whether the property is settable (true) or readonly (false)	true
+                        // or false	Yes	No (false)
+                        CPlugin_014_sendMQTTnode(nodename,
+                                                 deviceName,
+                                                 ExtraTaskSettings.TaskDeviceValueNames[varNr],
+                                                 F("/$settable"),
+                                                 F("true"),
+                                                 errorCounter);
+
+                        // $name	Device → Controller	Friendly name of the property.	Any String	Yes	No ("")
+                        valueName  = F("Homie Receiver: ");
+                        valueName += ExtraTaskSettings.TaskDeviceValueNames[varNr];
+                        CPlugin_014_sendMQTTnode(nodename,
+                                                 deviceName,
+                                                 ExtraTaskSettings.TaskDeviceValueNames[varNr],
+                                                 F("/$name"),
+                                                 valueName,
+                                                 errorCounter);
+
+                        // $datatype	The data type. See Payloads.	Enum: [integer, float, boolean,string, enum, color]
+                        unitName = F("");
+
+                        switch (Settings.TaskDevicePluginConfig[x][varNr]) {
+                          case 0: valueName = F("integer");
+
```

Comment:
switch/case without curly braces? Looks a bit odd... 😉

Reply:
Strictly speaking a case does only need to be wrapped in `{}` if you declare new members.
Then you really MUST use braces.